net/local/local_sockif.c

```diff
@@ -833,7 +833,7 @@ static int local_close(FAR struct socket *psock)
 static int local_ioctl(FAR struct socket *psock, int cmd, unsigned long arg)
 {
   FAR struct local_conn_s *conn;
-  int ret = OK;
+  int ret = -ENOTTY;
```

Comment:
@xiaoxiang781216 @anchao could you please take a look? I recall some changes around `FIONBIO` made some time ago related to default behavior.

Reply:
it's already be done at line 878.

Reply:
> it's already be done at line 878.

I do not think it is the same. If you see the first case FIONBIO, I think there is a way to go through this case without setting ret variable, thus ret was set to OK at init and this is not the proper return value in that case. Am I wrong here?
In nuttx-apps repo, I have submitted a PR for nng with Pub/Sub example and you can test it.

Reply:
Oh, in most case, lc_infile/lc_outfile shouldn't NULL.

Reply:
I do not remember exactly, because it was quite some time I did the debugging. I will try again the nng example this evening with the latest code and see what happens.

Reply:
Thanks.

Reply:
> Oh, in most case, lc_infile/lc_outfile shouldn't NULL. In your case, is both NULL?

Yes, Both are NULL. Here is my callstack:
![image](https://user-images.githubusercontent.com/2845923/223210808-69ee4d9f-603a-4aba-9b1a-7e4b7f9635d2.png)

Reply:
@xiaoxiang781216 so should we remove the assignment in the default case?

Reply:
Ok, let's remove line 877-879 which doesn't need anymore.
BTW, is it a listen socket? it make sense that both in/out pipe doesn't exist. What's difference return OK not -ENOTTY.

Reply:
Yes, is a listen socket. NNG code is calling (void) fcntl(fd, F_SETFL, O_NONBLOCK) to set the socket in non block mode. You can see from callstack above the call to fcntl. When local_ioctl returns ENOTTY, netdev_file_ioctl is called next, which is then responsible to set s_flags for the socket. In case of OK, this doesn't happen.

Reply:
Ok, the patch looks good, please remove the default case. BTW, https://github.com/apache/nuttx/pull/8751 will address the problem.

Reply:
Will test with NNG. 
Yesterday, when testing NNG PubSub example, I have found out that example doesn't work as it was a few months ago. Although I am getting to nng_recv (before that ENOTTY code blocked in dial) function in client thread, nothing seems to get through, altough server thread is pushing messages. Something must have been changed in local socket implementation?